### PR TITLE
Remove sri hashes

### PR DESCRIPTION
**Pull request by LnL7**
###### Motivation for this change

These break compatibility with nix 2.0, see #89275. Alternative would be to bump minvers.nix, but some of this might also be relevant for 20.03 and 19.09.

###### Things done

Validated using `nix-env -qa --out-path -f.` with new hash types disabled.

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by Ma27 — APPROVED** [review]

**Review by zimbatm — APPROVED** [review]

**Comment by vcunat** [comment]
Lowest release with SRI support is 2.2 (Jan 2019), if I look right, which is not that long.  Perhaps `minver.nix` should contain some hint about the lag we'll maintain here, e.g. it might say that we want at least two years margin.

Another issue is how to maintain the state; my suggestion: https://github.com/NixOS/ofborg/issues/513

**Review by FRidh — APPROVED** [review]

File: pkgs/development/tools/poetry2nix/poetry2nix/lib.nix [hunk 1]
```diff
@@ -146,7 +146,7 @@ let
     else
       pkgs.fetchurl {
         url = predictURLFromPypi { inherit pname file hash kind; };
-        inherit hash;
+        sha256 = builtins.elemAt (builtins.match "sha256:(.*)" hash) 0; # nix 2.0 backwards compatibility.
```
Comment (LnL7) [inline pkgs/development/tools/poetry2nix/poetry2nix/lib.nix:149]: If we go this route, poetry2nix should probably be updated to handle this properly.

/cc @adisbladis

**Review by srhb — APPROVED** [review]

**Comment by rycee** [comment]
Would it be worth also removing these from the release-20.03 branch as well?

**Comment by Ma27** [comment]
IMHO yes.

**Comment by LnL7** [comment]
Yeah, but I'm working a followup for base64 hashes first.

**Comment by edolstra** [comment]
@vcunat I disagree. I think in the past our rule was it should be possible to evaluate Nixpkgs using the Nix version from the previous NixOS release. So using SRI hashes is fine.

**Comment by edolstra** [comment]
https://github.com/NixOS/nixpkgs/pull/89476

**Comment by vcunat** [comment]
:thinking: oh, in terms of NixOS release support cycles it **is** a relatively long time.  Last default nix < 2.2 was in 18.09, which has been _unsupported_ for over a year now.

**Comment by bhipple** [comment]
What's the final takeaway here?  Should we allow SRI hashes in master now?  Base64?  And to clarify, the right format is this, right?

```nix
hash = "sha256-rJbpoPjNMlw4diWjwNQ/DPo3rltvISU4kuRqBbvlBZ0=";
```

**Comment by LnL7** [comment]
Master requires nix 2.2 now so using SRI hashes there is fine.

**Comment by bhipple** [comment]
Sounds good.  Should we revert your PR to get rid of base64 hashes?  I actually quite like them, as aside from being shorter they're also an external standard, unlike our homebrew base32 encoding scheme.

**Pull request merged**